osquery/tables/system/darwin/mdfind.mm

```diff
@@ -0,0 +1,119 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <CoreServices/CoreServices.h>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+typedef std::pair<MDQueryRef, std::string> NamedQuery;
+
+const size_t kMaxQueryWait = 5;
+
+void genResults(const std::vector<NamedQuery>& queries, QueryData& results) {
+  // The results can update live from macOS so we stop subscribing to updates
+  // so we have a moment in time
+  for (const auto& query : queries) {
+    MDQueryDisableUpdates(query.first);
+  }
+
+  // Get data from all the queries
+  for (const auto& query : queries) {
+    for (auto i{0}; i < MDQueryGetResultCount(query.first); ++i) {
+      MDItemRef mdi = (MDItemRef)(MDQueryGetResultAtIndex(query.first, i));
```

Comment:
Can this not be `auto mdi = static_cast<MDItemRef>(MDQueryGetResultAtIndex(query.first, i));`? Or `reinterpret`?